### PR TITLE
Try to fix coverall stats

### PR DESCRIPTION
**Pull request by FranzForstmayr**

### CODE REVIEW

**Comment by jhillairet** [comment]
OK I see... So, what is the point of the "--cov-append" option? Would be nice to combine the coverage of the notebooks and the one of the tests, if possible.

**Comment by FranzForstmayr** [comment]
Now all tests are run within a single tox session. Is this fine? 
The duration is slightly longer of course, however, it's easier for developers to test both tests and notebooks.

**Comment by jhillairet** [comment]
Few comments to be discussed:
- with the latest commit, running `pytest` also run the tests for all the notebooks, which takes of course muuuch more time. This is what bothers me the most: I think this is not very convenient for daily development, when users want to launch often the test suite on their computer to check their code... What do you think?
- the solution you found in 08a1594 of inverting the order of the test (putting the tests of the code at last) [also worked](https://coveralls.io/builds/44249987). In such case, I understand that the tests of the notebooks are not included in the coverage (61% vs 75%). This solution would allow using the `pytest` command locally to test only the code test suite, but reduces a bit the coverage percentage.
- question : how does tox deals with this multiple python versions? in the tox.ini they are defined twice. On my computer, running tox only runs the tests with the python version it found and produces an error for the other versions.

**Comment by FranzForstmayr** [comment]
> * with the latest commit, running `pytest` also run the tests for all the notebooks, which takes of course muuuch more time. This is what bothers me the most: I think this is not very convenient for daily development, when users want to launch often the test suite on their computer to check their code... What do you think?

Yes, that's true. I'll see what's possible. 

> * the solution you found in [08a1594](https://github.com/scikit-rf/scikit-rf/commit/08a15941e655cceb306aa738deeeae5758bd455b) of inverting the order of the test (putting the tests of the code at last) [also worked](https://coveralls.io/builds/44249987). In such case, I understand that the tests of the notebooks are not included in the coverage (61% vs 75%). This solution would allow using the `pytest` command locally to test only the code test suite, but reduces a bit the coverage percentage

`pytest` alone still works as expected, however if you have `tox` installed, you can run the tests again multiple python versions and the installer is tested too. 

> * question : how does tox deals with this multiple python versions? in the tox.ini they are defined twice. On my computer, running tox only runs the tests with the python version it found and produces an error for the other versions.
`tox` uses the python versions specified in the `envlist` if they are installed. Unfortunately github runner specifies python3.9 as `py3.9` whereas `tox` uses `py39`. `tox-github-actions` allows to create a translation table.

**Comment by FranzForstmayr** [comment]
Coverage seems to be pretty good, as the `vi` part is excluded from statistics.
Maybe we can find a solution to test the `vi` functions with mocked components later.

File: .github/workflows/testing.yml [hunk 1]
```diff
@@ -38,23 +30,12 @@ jobs:
...
         flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
-
-    # tests suites & creates coverage report
-    - name: Test with pytest
-      if: ${{ always() }}    
-      run: |
-        pytest
-
+  
     # test notebooks
```
Comment (jhillairet) [inline .github/workflows/testing.yml:34]:
```suggestion
    # test code and notebooks (tutorials and examples)
```

File: .github/workflows/testing.yml [hunk 2]
```diff
@@ -38,23 +30,12 @@ jobs:
...
-    - name: Run tutorials notebooks with nbval
-      if: ${{ always() }}    
-      run: |
-        python -m pytest --nbval-lax --current-env -p no:python doc/source/tutorials/
-
-    - name: Run examples notebooks with nbval
+    - name: Run tutorial & example notebooks with nbval
```
Comment (jhillairet) [inline .github/workflows/testing.yml:35]:
```suggestion
    - name: Test the code, tutorials and examples
```

File: tox.ini [hunk 3]
```diff
@@ -1,25 +1,32 @@
...
 commands =
     pytest {posargs}
 
 [pytest]
 testpaths = 
     skrf
-addopts = "--cov=skrf"
+    doc/source/examples
+    doc/source/tutorials
+addopts = --cov
```
Comment (jhillairet) [inline tox.ini:25]: I don't understand why the command `pytest` test only the `skrf` directory despite the doc is also indicated, but it works... ! ;) could you explain to me this magic ?

**Comment by jhillairet** [comment]
> Coverage seems to be pretty good, as the vi part is excluded from statistics.
>Maybe we can find a solution to test the vi functions with mocked components later.

OK, super nice. We will see for the vi module later, as it requires anyhow a full update per se.

**Review by jhillairet — APPROVED** [review]

File: .github/workflows/testing.yml (hunk 1 above)
Comment (jhillairet) [reply .github/workflows/testing.yml:34]: just to update the comments

File: tox.ini (hunk 3 above)
Comment (FranzForstmayr) [reply tox.ini:25]: I'm not hundred percent sure, I think `.ipynb` files aren't even recognized without setting the option `--nbval`.

Comment (FranzForstmayr) [reply tox.ini:25]: This is the regex pytest is using by default.
[pytest ini parser](https://github.com/pytest-dev/pytest/blob/062d91ab474881b58ae1ff49b4844402677faf45/src/_pytest/python.py#L108)

Comment (jhillairet) [reply tox.ini:25]: OK. So the test of the notebook is performed via tox because of the `nbval` option.

**Comment by FranzForstmayr** [comment]
Remove --cov-append as it isn't necesary anymore. I'm fine with this PR :)

**Comment by jhillairet** [comment]
fixes #565

**Pull request merged**